sympy/printing/smtlib.py

```diff
@@ -0,0 +1,515 @@
+import builtins
+import typing
+
+import sympy
+from sympy.core import Add, Mul
+from sympy.core import Symbol, Expr, Float, Rational, Integer, Basic
+from sympy.core.function import UndefinedFunction, Function
+from sympy.core.relational import Relational, Unequality, Equality, LessThan, GreaterThan, StrictLessThan, StrictGreaterThan
+from sympy.functions.elementary.complexes import Abs
+from sympy.functions.elementary.exponential import exp, log, Pow
+from sympy.functions.elementary.hyperbolic import sinh, cosh, tanh
+from sympy.functions.elementary.miscellaneous import Min, Max
+from sympy.functions.elementary.piecewise import Piecewise
+from sympy.functions.elementary.trigonometric import sin, cos, tan, asin, acos, atan, atan2
+from sympy.logic.boolalg import And, Or, Xor, Implies
+from sympy.logic.boolalg import BooleanTrue, BooleanFalse, BooleanFunction, Not, ITE
+from sympy.printing.printer import Printer
+from sympy.sets import Interval
+
+
+class SMTLibPrinter(Printer):
+    printmethod = "_smt"
+
+    # based on dReal, an automated reasoning tool for solving problems that can be encoded as first-order logic formulas over the real numbers.
+    # dReal's special strength is in handling problems that involve a wide range of nonlinear real functions.
+    _default_settings = {
+        'precision': 5,
```

Comment:
Why 5 here (and 16 below)? I think the default precision should be None, which will just use the default precision of evalf (when no precision is passed to it).

Reply:
Oops, my mistake!  I like the None idea, it is very clean. Will fix!